code/modules/cooking/machinery/cooking_machines/container.dm

```diff
@@ -0,0 +1,264 @@
+	icon_state = "skillet"
+	volume = 15
+	force = 11
+	hitsound = 'sound/weapons/smash.ogg'
+	flags = OPENCONTAINER // Will still react
+	appliancetype = SKILLET
+
+/obj/item/reagent_containers/cooking_container/skillet/Initialize(var/mapload, var/mat_key)
+	. = ..(mapload)
+	var/material/material = SSmaterials.get_material_by_name(mat_key || MATERIAL_STEEL)
+	if(!material)
+		return
+	if(material.name != MATERIAL_STEEL)
+		color = material.icon_colour
+	name = "[material.display_name] [initial(name)]"
+
+/obj/item/reagent_containers/cooking_container/saucepan
+	name = "saucepan"
+	shortname = "saucepan"
+	desc = "Is it a pot? Is it a pan? It's a saucepan!"
+	icon_state = "pan"
+	volume = 60
+	slot_flags = SLOT_HEAD
+	force = 8
+	hitsound = 'sound/weapons/smash.ogg'
+	flags = OPENCONTAINER // Will still react
+	appliancetype = SAUCEPAN
+
+/obj/item/reagent_containers/cooking_container/saucepan/Initialize(var/mapload, var/mat_key)
+	. = ..(mapload)
+	var/material/material = SSmaterials.get_material_by_name(mat_key || MATERIAL_STEEL)
+	if(!material)
+		return
+	if(material.name != MATERIAL_STEEL)
+		color = material.icon_colour
+	name = "[material.display_name] [initial(name)]"
+
+/obj/item/reagent_containers/cooking_container/pot
+	name = "cooking pot"
+	shortname = "pot"
+	desc = "Boil things with this. Maybe even stick 'em in a stew."
+	icon_state = "pot"
+	max_space = 50
+	volume = 180
+	force = 8
+	hitsound = 'sound/weapons/smash.ogg'
+	flags = OPENCONTAINER // Will still react
+	appliancetype = POT
+	w_class = ITEMSIZE_LARGE
+
+/obj/item/reagent_containers/cooking_container/pot/Initialize(mapload, mat_key)
+	. = ..(mapload)
+	var/material/material = SSmaterials.get_material_by_name(mat_key || MATERIAL_STEEL)
+	if(!material)
+		return
+	if(mat_key && mat_key != MATERIAL_STEEL)
+		color = material.icon_colour
+	name = "[material.display_name] [initial(name)]"
+
+/obj/item/reagent_containers/cooking_container/fryer
+	name = "fryer basket"
+	shortname = "basket"
+	desc = "Put ingredients in this; designed for use with a deep fryer. Warranty void if used."
+	icon_state = "basket"
+	appliancetype = FRYER
+
+/obj/item/reagent_containers/cooking_container/plate
+	name = "serving plate"
+	shortname = "plate"
+	desc = "A plate. You plate foods on this plate."
+	icon_state = "plate"
+	appliancetype = MIX
+	volume = 5 // for things like jelly sandwiches etc
+
+/obj/item/reagent_containers/cooking_container/plate/MouseDrop(var/obj/over_obj)
+	if(over_obj != usr || use_check(usr))
+		return ..()
+	if(!(length(contents) || reagents?.total_volume))
+		return ..()
+	var/decl/recipe/recipe = select_recipe(src, appliance = appliancetype)
+	if(!recipe)
+		return
+	var/list/results = recipe.make_food(src)
+	var/obj/temp = new /obj(src) //To prevent infinite loops, all results will be moved into a temporary location so they're not considered as inputs for other recipes
+	for (var/atom/movable/AM in results)
+		AM.forceMove(temp)
```

Comment:
```suggestion
	for (var/result in results)
		var/atom/movable/AM = result
		AM.forceMove(temp)
```

Also elsewhere in this file.

Reply:
what IS it with you and micro-optimisations